lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1373,6 +1402,13 @@ void SessionState::set_monitor(
   monitor_map_[key] = std::make_unique<Monitor>(monitor);
 }
 
+bool SessionState::should_delete_monitor(SessionCredit& credit){
```

Comment:
Can we attach this function to the Monitor struct? Also we can just do `return credit.current_grant_contains_zero() && credit.is_quota_exhausted(1);` here right?

Reply:
i forgot about monitor structure. Good catch.
BTW, i put the logic in .h file. Not sure if that is a good practice or not🤫

Reply:
Yeah we can move that out to its own CPP/H. I think it's probably not the best practice, as if the header is included in two places it might cause problems. (not 100% sure). :D